### DISCUSSION TITLE
Stutter every few seconds

### DISCUSSION THREAD

**Discussion by bilbohob** [KnuckleCracker/CW4-bug-tracker#354] [General]
Hello,

I experience stutters every few second when the game is running (in a map with the game not paused)

I have a good config : 
CPU I7-9700K @ 3.600 GHz 8 CPU
GeForce RTX 2060 6go VRAM
16 GO RAM

The game is not umplayable but it is very frustationg.

Thank you very much.
[DxDiag.txt](https://github.com/KnuckleCracker/CW4-bug-tracker/files/5641888/DxDiag.txt)

**Comment by knucracker** [comment]
So when the game is paused, there is no stutter. When the game it running, that's the only time you see the stutter?  How would you characterize the stutter? Is is a long pause, a barely noticeable pause, etc.
In your dxdiag I see you are running on a 4k monitor at 60hz. Is the game set to run at 4k or do you have it at a lower resolution? Also is the game in a window or fullscreen?

Try varios combinations of window, lower res, etc. and see if that affected the stutter. Let me know what you find...

**Comment by Karsten75** [comment]
I messed up earlier when adding the "duplicate tag" 

This seems a duplicate of the issues in #350 after the earlier issues in that thread were resolved

**Comment by bilbohob** [comment]
Hello,

The stutter is regular and last for one second. It's more like a frame drop than a freeze.
The frame drop doesn't register with the steam FPS counter.
I tried changing the resolution to lower and reducing the graphics quality but nothing helps.
I also tried in windowed and in full-screen but it doesn't help either.

Thank you for your time, I continued the game and it's great :)

**Comment by Karsten75** [comment]
We recently had more reports of "stuttering. In one case, there was associated high power use (200W) from the GPU. The user had a large number (<200) chrome tabs.

They updated their driver and rebooted to clean things out. That resolved their issue and power consumption dropped to approx 40W.

If you can perhaps take a video (with your phone) that demonstrates the stuttering.

Also, please try this. Zoom in as close as you can, move the camera as far left as it can go. Using arorw keys, navigate as far left as you can go. Do you see the stuttering then?

Again, if you do, please attempt to take a video and upload that.

**Comment by bilbohob** [comment]
Hello,

I'm sorry but my phone is not powerful enough to capture the frame change.
I captured it with NVIDIA overlay.

Beside the occasional stuttering of the capture you can see it clearly at the middle of the video. It does it even when paused.

I exited Mozilla (with 4-5 tabs) and all others apps but it didn't change anything.

The video.
https://drive.google.com/file/d/1RFr3TwubqAIytlBzOTC-N_-Q3y6nEDFJ/view?usp=sharing

Thank you.

**Comment by Karsten75** [comment]
Might well be the overlays? We have not been able to recreate any stutter on any machine we have access to.

**Comment by bilbohob** [comment]
I have tried disabling the steam and Nvidia overlay. I also disabled f.lux and put the game in 1920*1080 but nothing is working.
My drivers are updated. I also put the nvidia 3d settings at default, it didn't help.

Thank you.

**Comment by Karsten75** [comment]
I am out of ideas, I will ask @knucracker  to take another look before we close this as something we can't recreate and have insufficient supporting material to attempt a resolution.

**Comment by ZoidtheVoid** [comment]
I have exactly the same issue. Game stutters every 3 seconds, no matter what quality settings/resolution I use. It also happens while paused. I've closed every other program, no overlays, using the gog version.
GPU drivers are up to date. 
CPU: Ryzen 3600x
GPU: Nvidia 1080
16 GB RAM
I didn't have this issue in the demo version.

**Comment by knucracker** [comment]
Is game mode on?  Try turning it either on or off depending on your current state and see if that makes a difference.
Click the start menu, then the settings gear, then Game Mode.
This is one reference article regarding game mode: https://www.techradar.com/news/windows-10-users-should-turn-this-feature-off-now-for-better-gaming-performance

**Comment by ZoidtheVoid** [comment]
No, game mode is disabled. Enabling it did no difference.

**Comment by knucracker** [comment]
And the stutter is a long stutter. The earlier reporter mention a stutter that would last 1 second. Is that the case for you as well? or is it more of a quicker thing (a jitter, etc.)  You mentioned the stutter was regardless of settings/resolution. Is that also true for full screen vs window? Also, does the length of the stutter change if the window is small, the game is paused, etc. And lastly, is you system a 4k system, multi monitors, etc?

**Comment by ZoidtheVoid** [comment]
The stutter lasts for roughly half a second at a 3 second interval.
It happens both in windowed and fullscreen regardless of the resolution/window size and it always stays the same; half a second every 3 seconds. It also happens while the game is paused. No matter what's being displayed in the game or what settings I use, it always stays the same.
I'm using a dual monitor setup with a 1440p and a 1080p monitor.

**Comment by knucracker** [comment]
Does it happen on the main menu? If you watch the planets rotate, do they stop,etc.  Also, this is a big ask I know, but does it happen if you only have one monitor on/connected?  A few people have mentioned the stutter problem, but most folks don't seem to have it. I've not been able to repro it, and none of the beta testers did either. Looking around at "stutter" issues for other games most seem to be centered around nvidia (but not all). I'm running nvidia on my main desktop, though, but don't have any stutter. So I'm looking for what might be 'different' or unique. For what it's worth, I have two monitors as well. Both are 60HZ (a 1440 and a 1080 like you).

**Comment by ZoidtheVoid** [comment]
No, it doesn't happen in the main menu.
And yes, it still happens with only a single monitor connected to the GPU.
I've just realized that this stutter only happens when moving the camera. I'm not even sure if it's the game that's stuttering, or just the camera. The game runs perfectly smooth as long as I don't move the camera.
Edit: After closer examination it looks like I was wrong; the stutter happens regardless of camera movement, it's just very difficult to notice.

**Comment by knucracker** [comment]
That's an interesting observation.  Do you move the camera with the mouse or keyboard? (And have you remapped any keys).  And there is camera panning (right left up down) and then there is map zooming. There is also free cam (press shift-f by default).  See if  you can draw any observations about any of these options

**Comment by ZoidtheVoid** [comment]
I've used the free camera to validate my previous statement about the camera, and the stutters are definitely noticable while close up and they happen even when the camera is stationary. The "standard" camera distance is simply too far away for me to notice any stutters in the animations without moving the camera, so I can't say much about that.  
So my previous statement about the camera being involved was wrong, I simply didn't test it thoroughly enough.
If there's anything else I can try out please let me know.

**Comment by bilbohob** [comment]
I have tested the same things as ZoidtheVoid and found the same problems.
I am at your disposal for any additional information.
For information my screen setup is 4k for main screen and 1080p for second screen.

**Comment by knucracker** [comment]
Ok, so no mouse movement necessary.  Try this: Create a new map using the map editor. Average size is fine. It will come up as a flat blank map. Land the Rift Lab in the middle of the map and zoom in a bit so you can see the spot light rotate around at the top of the rift lab. Unpause and just let the game run. Don't move or pan the map. Watch the spotlight rotate and see if that stutters every 3 seconds.

If it does, then go to the settings performance tab and uncheck "High FPS" and then come back and watch the spotlight for 30 seconds or so. If that doesn't make a difference, then turn High FPS back on.

Try moving the Rift lab to a corner of the map then pan the map so that you don't see much of the map, but you can still see the rift lab. Zoom in so most of the map is not visible. See if that affects the stutter of the rotating light.

Then take a look at system performance. Bring up the task manager and watch processes and GPU usage to see if you can determine if there is a periodic spike by anything.  If you have TechPowerUp GPU-Z installed, bring that up and watch GPU load to see if there is a spike by anything. On the sensors tab of GPUZ there is a setting button in the top right. In settings there is a sensors tab that has a way to increase the refresh rate.  There is also a "log to file" toggle on GPUZ (bottom left of the sensors tab) that will allow you to log the readings to a file (just uncheck to stop logging).

I'm looking for 'something' that could possibly cause a stutter. The game itself, on a blank map doesn't really do much (especially when paused).

**Comment by Bob2Chiv** [comment]
one user (perhaps one of the people on this tracker) reported the following in Discord:
"Looks like it was iCue somehow causing my stutters... Quit the app and now it's running just fine... "
https://discordapp.com/channels/192420539204239361/776875860501463061/785722679800889365

**Comment by ZoidtheVoid** [comment]
I followed your instructions, but like I said before, I find it very difficult to notice any stutters in the animations because of the camera distance. I attached 2 logs to this post; both of them are on a blank map with average size with the Rift Lab in the middle of the map. Camera was zoomed in to the maximum, with the Rift Lab in the center. 
[60 fpsGPU-Z Sensor Log.txt](https://github.com/KnuckleCracker/CW4-bug-tracker/files/5660943/60.fpsGPU-Z.Sensor.Log.txt)
[30fpsGPU-Z Sensor Log.txt](https://github.com/KnuckleCracker/CW4-bug-tracker/files/5660944/30fpsGPU-Z.Sensor.Log.txt)

**Comment by knucracker** [comment]
I've converted this issue into a discussion to attract a wider audience for help.

**Comment by Karsten75** [comment]
Another data point from a Discord contributor:  ( https://discord.com/channels/192420539204239361/776875860501463061/787717464586584115 )

One thing I have noticed in case it helps other - I keep a controller plugged in but the connection is a little spotty - when the controller disconnects/reconnects it causes a half second stutter. So my stuttering issues were fixed by unplugging my controller while playing. I suspect it's a problem with the steam controller API, as it effects other games as well - had the same issue in Monster Sanctuary.

**Comment by BilboGCL** [comment]
Btw, with my new laptop, I'm in this club, too... Added the info text in file:
[info.txt](https://github.com/KnuckleCracker/CW4-bug-tracker/files/6065543/info.txt)
Here it feels like half a second maybe every ten or twenty seconds, but hard to say as I often play paused and while it also appears in pause, it doesn't disturb me that much.
Hardware things that have changed: i7-10750H, 1650TI, probably not related, but now its a killer WiFi card
Software things that have changed (and I'm aware of): NVIDIA driver is actually including the overlay thing,

**Reply by knucracker** [reply]
Try running at 1920x1080 fullscreen _exclusive_ mode and see if that makes a difference.  Also, windows 10 game mode... try turning it on or off and see if either makes a difference.  If you have new laptop, also make sure you don't have some bundled software that could cause the slowdown.  Anything from trial AV software to audio software can wake up periodically and cause a system interruption.  If nothing make a difference you might also try watching the windows task manager / perf monitor task list (sort by cpu usage).  And see if some process is jumping up to the top or in 2nd place periodically.

**Comment by BilboGCL** [comment]
1st thing I did was your suggestion for the exclusive mode, that worked as a charm, no stuttering at all. While in exclusive mode, I disabled the NVidia overlay thing and switched back to non-exclusive mode as the exclusive mode has annoying side effects (alt+tab is no fun). And stuttering didn't reappear. Hope that helps somebody else!
